pytng/pytng.pyx

```diff
@@ -264,6 +331,64 @@ cdef class TNGFile:
         self.step += 1
         return TNGFrame(xyz, time, self.step - 1, box)
 
+    def _init_first_frame_write(self, int64_t n_atoms,
+                                box_shape_interval=1, pos_interval=1,
+                                time=None):
+        cdef int64_t ok
+
+        ok = tng_implicit_num_particles_set(self._traj, n_atoms)
+        if_not_ok(ok, 'Could not set the number of particles')
+
+        ok = tng_util_pos_write_interval_set(self._traj, 1)
+        if_not_ok(ok, 'Could not set position write interval')
+        ok = tng_util_box_shape_write_interval_set(self._traj, 1)
+        if_not_ok(ok, 'Could not set box shape write interval')
+
+    def write(self,
+              np.ndarray[np.float32_t, ndim=2, mode='c'] positions,
+              np.ndarray[np.float32_t, ndim=2, mode='c'] box,
+              time=None):
+        if self.mode != 'w':
+            raise IOError('File opened in mode: {}. Writing only allowed '
+                          'in mode "w"'.format(self.mode))
+        if not self.is_open:
+            raise IOError('No file currently opened')
+
+        cdef int64_t ok
+        cdef np.ndarray[float, ndim=2, mode='c'] xyz
+        cdef np.ndarray[float, ndim=2, mode='c'] box_contiguous
+
+        if self._n_frames == 0:
+            self._init_first_frame_write(n_atoms=positions.shape[0])
+
+        if time is not None:
+            try:
+                time = float(time)
+            except ValueError:
+                raise ValueError('time must be a real number or None')
+
+        box_contiguous = np.ascontiguousarray(box, dtype=np.float32)
+        if time is None:
+            ok = tng_util_box_shape_write(self._traj, self.step,
+                                          &box_contiguous[0, 0])
+        else:
+            ok = tng_util_box_shape_with_time_write(self._traj,
+                                                    self.step,
+                                                    time,
+                                                    &box_contiguous[0, 0])
+        if_not_ok(ok, 'Could not write box shape')
+
+        xyz = np.ascontiguousarray(positions, dtype=np.float32)
+        if time is None:
+            ok = tng_util_pos_write(self._traj, self.step, &xyz[0, 0])
+        else:
+            ok = tng_util_pos_with_time_write(self._traj, self.step,
+                                              time, &xyz[0, 0])
```

Comment:
why write the time twice? That seems strange. Maybe we need to ask in the gromacs devel list